lib/src/main/java/org/team100/lib/localization/NotePosition24ArrayListener.java

```diff
@@ -43,48 +48,83 @@ void consumeValues(NetworkTableEvent e) {
             } catch (RuntimeException ex) {
                 return;
             }
-
-            for (NotePosition24 position : positions) {
-                // this is where you would do something useful with the payload
-                // System.out.println(fields[1] + " " + position);
-            }
+            updateMeasurement(fields[1]);
         } else {
             System.out.println("note weird vision update key: " + name);
         }
     }
 
+    private Optional<Double> optionalX(double position) {
+        switch (Identity.instance) {
+            case BETA_BOT:
+            case COMP_BOT:
+                Optional<Double> e = Optional.of(position);
+                return e;
+            default:
+                return Optional.empty();
+        }
+    }
+
+    private Optional<Double> optionalY(double position) {
+        switch (Identity.instance) {
+            case BETA_BOT:
+            case COMP_BOT:
+                Optional<Double> e = Optional.of(position);
+                return e;
+            default:
+                return Optional.empty();
+        }
+    }
+
+    private void updateMeasurement(String ID) {
+        Transform3d cameraInRobotCoordinates = Camera.get(ID).getOffset();
+        CameraAngles e = new CameraAngles(cameraInRobotCoordinates);
+        double dx = positions[0].getYaw();
```

Comment:
looks like you're just using the first entry in the position array.  why?  does the python pick the best target as the first entry?

Reply:
To be honest, I was going to do better testing on this, I don't exactly know how to but I was planning on picking either the biggest contour, i'll add a todo to take care of it tomorrow